components/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt/src/main/java/org/wso2/carbon/identity/api/resource/mgt/APIResourceManagerImpl.java

```diff
@@ -82,7 +86,26 @@ public APIResource getAPIResourceById(String apiResourceId, String tenantDomain)
     public APIResource addAPIResource(APIResource apiResource, String tenantDomain)
             throws APIResourceMgtException {
 
-        return CACHE_BACKED_DAO.addAPIResource(apiResource, IdentityTenantUtil.getTenantId(tenantDomain));
+        try {
```

Comment:
Use this,

https://github.com/wso2/identity-organization-management-core/blob/main/components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/util/Utils.java#L402